packages/strapi-design-system/src/DismissableLayer/DimissableLayer.js

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+export const DismissableLayer = ({ children, className, onEscapeKeyDown, onPointerDownOutside }) => {
+  const layerRef = React.useRef(null);
+  const onEscapeKeyDownHandler = useCallbackRef(onEscapeKeyDown);
+  const onPointerDownOutsideHandler = useCallbackRef(onPointerDownOutside);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').KeyboardEventHandler<Document>}
+     */
+    const handleKeyDown = (event) => {
+      if (event.key === 'Escape') {
+        onEscapeKeyDownHandler(event);
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, [onEscapeKeyDownHandler]);
+
+  React.useEffect(() => {
+    /**
+     * @type {import('react').PointerEventHandler<Document>}
+     */
+    const handlePointerDownOutside = (event) => {
+      if (layerRef.current && !layerRef.current.contains(event.target)) {
+        onPointerDownOutsideHandler();
+      }
+    };
+
+    document.addEventListener('pointerdown', handlePointerDownOutside);
+
+    return () => document.removeEventListener('pointerdown', handlePointerDownOutside);
+  }, [onPointerDownOutsideHandler]);
+
+  return (
+    <div ref={layerRef} className={className}>
+      {children}
+    </div>
+  );
+};
+
+DismissableLayer.propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
+  onEscapeKeyDown: PropTypes.func,
+  onPointerDownOutside: PropTypes.func,
+};
+
+/**
+ * A custom hook that converts a callback to a ref to avoid triggering re-renders when passed as a
+ * prop or avoid re-executing effects when passed as a dependency
+ *
+ * Stolen from @radix-ui/react-use-callback-ref
+ */
+function useCallbackRef(callback) {
```

Comment:
do you think it would be worth adding it to DS utils? @joshuaellis

Reply:
I haven't used it in other components, so I suppose the question is "is there a need for it"? Happy to take the consensus of the group 👍🏼

Reply:
As long as we only need it here, I'm okay with keeping it local. Tests (for the whole component) would be very nice though :)